### PR TITLE
Transformers

### PR DESCRIPTION
**Pull request by philecs**
Added an extra document for transformation language and expanded the section "transformer evolution"

### CODE REVIEW

File: Transformer/README.md [hunk 1]
```diff
@@ -5,26 +5,37 @@ A transformer is responsible to semantically transform an incoming JSON object i
...
 
 ## Transformer Evolution
-Sometimes there is a need to change the existing data model to adjust to different requirements. In this case, a transformer needs to be adjusted/updated in order to be compatible with the newest version of the data model (Backward compatibility is supported, therefore an existing transformer can still be used with previous versions of the model).
+Sometimes there is a need to change the existing data model to adjust to different requirements. In this case, a transformer needs to be adjusted/updated in order to be compatible with the newest version of the data model.
+
+Some changes do not affect existing mappings/transformations and transformers can still be used (although they are not reffering to the newest model version). Thus, backward compatibility is automatically given. According to the [OData specification](http://docs.oasis-open.org/odata/odata/v4.0/errata03/os/complete/part1-protocol/odata-v4.0-errata03-os-part1-protocol-complete.html#_Toc453752210) (shortened/modified list) the following changes do not require a change within a transformer:
```
Comment (jhorbulyk) [inline Transformer/README.md]: `referring`

File: Transformer/TransformationLanguage.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+Short description of model transformation languages in general:
+
+> A model transformation, which is essentially a program which operates on models, can be written in a general-purpose programming language, such as Java. However, special-purpose model transformation languages can offer advantages, such as syntax that makes it easy to refer to model elements.
+
+_Source:_ [Wikipedia](https://en.wikipedia.org/wiki/Model_transformation_language)
+
+Due to its focus on JSON objects and its simplicity, JSONata is the recommended transformation language for creating a transformer.
+
+# JSONata Overview
```
Comment (jhorbulyk) [inline Transformer/TransformationLanguage.md]: Adding a link to the JSONata documentation and trial site would be helpful here. (As opposed to at the bottom.)

File: Transformer/TransformationLanguage.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+- Is a Lightweight query and transformation language for JSON data
+- Helps to extract specific meaningful data from JSON data using a simple logic
+- Supports more than 20 operators
+- Includes numerous functions such as string-, numeric-, array- and object-functions
+
+## Exemplary Functions
```
Comment (jhorbulyk) [inline Transformer/TransformationLanguage.md]: More examples can be seen at https://support.elastic.io/support/solutions/articles/14000069448-jsonata-powered-mapper

Comment (philecs) [reply Transformer/TransformationLanguage.md]: Good point. I'll have a look if I can incorporate it.

File: Transformer/TransformationLanguage.md (hunk 2 above)
Comment (philecs) [reply Transformer/TransformationLanguage.md]: Makes sense. Also wanted to add the JSONata tester.

File: Transformer/README.md (hunk 1 above)
Comment (philecs) [reply Transformer/README.md]: Was your question answered during the call?

Comment (jhorbulyk) [reply Transformer/README.md]: I think so.

**Pull request merged**